### PR TITLE
Change enum usages to `enum | int`

### PR DESCRIPTION
**Pull request by SweetMNM**
Can't automatically merge since i forgot to pull before

### CODE REVIEW

File: typings/Entity.d.ts [hunk 1]
```diff
@@ -48,10 +48,10 @@ declare class Entity {
   IsActiveEnemy(includeDead: boolean): boolean;
   IsVulnerableEnemy(): boolean;
   IsFlying(): boolean;
-  AddEntityFlags(entityFlags: EntityFlag): void;
-  ClearEntityFlags(entityFlags: EntityFlag): void;
-  GetEntityFlags(): EntityFlag;
-  HasEntityFlags(entityFlags: EntityFlag): boolean;
+  AddEntityFlags(entityFlags: EntityFlag | int): void;
```
Comment (Zamiell) [inline typings/Entity.d.ts:51]: this is incorrect, because there is no such thing as a custom EntityFlag

File: typings/Entity.d.ts [hunk 2]
```diff
@@ -96,10 +96,10 @@ declare class Entity {
   Friction: float;
   Position: Vector;
   Velocity: Vector;
-  readonly Type: EntityType;
-  Variant: EntityVariantForAC;
+  readonly Type: EntityType | int;
+  Variant: EntityVariantForAC
```
Comment (Zamiell) [inline typings/Entity.d.ts:100]: should be EntityVariantForAC | int

File: typings/Entity.d.ts [hunk 3]
```diff
@@ -115,8 +115,8 @@ declare class Entity {
   HitPoints: float;
   readonly Index: int;
   readonly TargetPosition: Readonly<Vector>;
-  GridCollisionClass: EntityGridCollisionClass;
-  EntityCollisionClass: EntityCollisionClass;
+  GridCollisionClass: GridCollisionClass | int;
```
Comment (Zamiell) [inline typings/Entity.d.ts:118]: you introduced a bug here

File: typings/Entity.d.ts [hunk 4]
```diff
@@ -115,8 +115,8 @@ declare class Entity {
...
+  EntityCollisionClass: EntityCollisionClass | int;
```
Comment (Zamiell) [inline typings/Entity.d.ts:119]: this is incorrect, there are no custom EntityCollisionClass

File: typings/EntityBomb.d.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 declare class EntityBomb extends Entity {
   SetExplosionCountdown(countdown: int): void;
 
-  Flags: TearFlags;
+  Flags: TearFlags | int;
```
Comment (Zamiell) [inline typings/EntityBomb.d.ts:4]: incorrect, there are no custom TearFlags

File: typings/EntityFamiliar.d.ts [hunk 6]
```diff
@@ -8,9 +8,9 @@ declare class EntityFamiliar extends Entity {
   GetOrbitPosition(position: Vector): Vector;
   Shoot(): void;
   FireProjectile(direction: Vector): EntityTear;
-  PlayChargeAnim(direction: Direction): void;
-  PlayShootAnim(direction: Direction): void;
-  PlayFloatAnim(direction: Direction): void;
+  PlayChargeAnim(direction: Direction | int): void;
```
Comment (Zamiell) [inline typings/EntityFamiliar.d.ts:11]: incorrect, there are no custom Direction

File: typings/EntityNPC.d.ts [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
-    championColorIdx: ChampionColorIdx,
+    championColorIdx: ChampionColorIdx | int,
   ): boolean;
   KillUnique(): void;
   IsBoss(): boolean;
   GetPlayerTarget(): Entity;
   CalcTargetPosition(distanceLimit: float): Vector;
   GetBossColorIdx(): int;
-  GetChampionColorIdx(): ChampionColorIdx;
+  GetChampionColorIdx(): ChampionColorIdx | int;
```
Comment (Zamiell) [inline typings/EntityNPC.d.ts:13]: incorrect, there are no custom ChampionColorIdx

File: typings/EntityNPC.d.ts [hunk 8]
```diff
@@ -28,7 +28,7 @@ declare class EntityNPC extends Entity {
   FireProjectiles(
     position: Vector,
     velocity: Vector,
-    projectilesMode: ProjectilesMode,
+    projectilesMode: ProjectilesMode | int,
```
Comment (Zamiell) [inline typings/EntityNPC.d.ts:31]: incorrect, there are no custom ChampionColorIdx

File: typings/Level.d.ts [hunk 9]
```diff
@@ -1,19 +1,19 @@
...
     dyslexia: boolean,
   ): string;
-  CanStageHaveCurseOfLabyrinth(levelStage: LevelStage): boolean;
+  CanStageHaveCurseOfLabyrinth(levelStage: LevelStage | int): boolean;
   GetCurseName(): string;
   ShowName(sticky: boolean): void;
-  GetStateFlag(levelStateFlag: LevelStateFlag): boolean;
-  SetStateFlag(levelStateFlag: LevelStateFlag, val: boolean): void;
+  GetStateFlag(levelStateFlag: LevelStateFlag | int): boolean;
+  SetStateFlag(levelStateFlag: LevelStateFlag | int, val: boolean): void;
```
Comment (Zamiell) [inline typings/Level.d.ts:16]: same with LevelStateFlag

File: typings/TemporaryEffects.d.ts [hunk 10]
```diff
@@ -1,19 +1,19 @@
...
+  RemoveNullEffect(nullItemID: NullItemID | int): void;
+  HasCollectibleEffect(collectibleType: CollectibleType | int): boolean;
+  HasTrinketEffect(trinketType: TrinketType | int): boolean;
+  HasNullEffect(nullItemID: NullItemID | int): boolean;
+  GetCollectibleEffect(collectibleType: CollectibleType | int): Readonly<TemporaryEffect>;
+  GetTrinketEffect(trinketType: TrinketType | int): Readonly<TemporaryEffect>;
+  GetNullEffect(nullItemID: NullItemID | int): Readonly<TemporaryEffect>;
+  GetCollectibleEffectNum(collectibleType: CollectibleType | int): int;
+  GetTrinketEffectNum(trinketType: TrinketType | int): int;
+  GetNullEffectNum(nullItemID: NullItemID | int): int;
```
Comment (Zamiell) [inline typings/TemporaryEffects.d.ts:16]: same with NullItemID

File: typings/Entity.d.ts (hunk 2 above)
Comment (SweetMNM) [reply typings/Entity.d.ts:100]: Oh i guess it got lost when fixing merge conflicts

File: typings/Entity.d.ts (hunk 3 above)
Comment (SweetMNM) [reply typings/Entity.d.ts:118]: I am not sure i understand what EntityGridCollisionClass is, should it not be `GridCollisionClass `?

**Pull request merged**